fms/modules/feedforward.py

```diff
@@ -123,8 +124,16 @@ def __init__(
             use_bias,
             gain,
         )
-        self.rank = rank
-        self.world_size = world_size
+        TPModule.setup_tp(self, rank, world_size)
```

Comment:
is this the same as `self.setup_tp(rank, world_size)`? seems odd to call it as a static function with `self` if so

Reply:
I know sometimes the diamond inheritance in python is a pain (I guess it's always a pain), and I was trying to make sure the right function was getting called, but it should work with `self`. I'll test and make the change if it works